### PR TITLE
add e2e tests for users/roles list view for standard user

### LINKED ISSUE #8777: Automation: List global roles/users pages for standard users when they have custom permissions assigned
Taken from work done in https://github.com/rancher/dashboard/issues/8344, we should write an automated test to ensure that we do list the appropriate material in Page Configuration >> Users & Authentication pages for a standard user who has the roles or users.

E2E feels like the goal

To reproduce:
1. As an admin, create a global role by navigating to Users and Auth >> Roles >> Global Roles >> Verbs: List, Get & Resources: Global Roles
2. Create a standard user as an admin and assign the role
3. Log in as standard user

Expected result:
The user to be able to list/get the pages `Users & Auth`

Factors:
- We need to add a test suite around non-admin users.

### PR DESCRIPTION
**Pull request by aalves08**
Fixes #8777 

- Adds e2e test

@richard-cox @scures had to redo the PR as the other one got bricked 🤷 -> https://github.com/rancher/dashboard/pull/9054

Basically it has the fixes suggested by @scures on https://github.com/rancher/dashboard/pull/9069 + the removal of the `wait` as per @richard-cox comment

### CODE REVIEW

File: cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
+import LabeledInputPo from '@/cypress/e2e/po/components/labeled-input.po';
+import LabeledSelectPo from '@/cypress/e2e/po/components/labeled-select.po';
+import AsyncButtonPo from '@/cypress/e2e/po/components/async-button.po';
+import CheckboxInputPo from '~/cypress/e2e/po/components/checkbox-input.po';
+
+export default class UsersAndAuthPo extends PagePo {
+  static url: string = '/c/_/auth/management.cattle.io.user'
```
Comment (richard-cox) [inline cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts:9]: This needs to match the pattern below. It's not an issue here, but will save someone getting confused if they write a test that goes via another cluster (the blank cluster will change to the last one visited)

```
  private static createPath(clusterId: string) {
    return `/c/${ clusterId }/manager/provisioning.cattle.io.cluster`;
  }

  static goTo(clusterId: string): Cypress.Chainable<Cypress.AUTWindow> {
    return super.goTo(ClusterManagerListPagePo.createPath(clusterId));
  }

  constructor(clusterId: string = '_') {
    super(ClusterManagerListPagePo.createPath(clusterId));
  }
```

File: cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return baseResourceList.resourceTable().sortableTable().rowElements();
+  }
+
+  listElementWithName(name: string) {
+    const baseResourceList = new BaseResourceList(this.self());
+
+    return baseResourceList.resourceTable().sortableTable().rowElementWithName(name);
+  }
+
+  listTitle() {
```
Comment (richard-cox) [inline cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts:37]: This should go in `ResourceListMastheadPo` (as it's part of the masthead component)

File: cypress/e2e/tests/pages/users-and-auth/users-and-auth.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    usersAdmin.username().set(standardUsername);
+    usersAdmin.newPass().set(standardPassword);
+    usersAdmin.confirmNewPass().set(standardPassword);
+    usersAdmin.selectCheckbox(customRoleName).set();
+    usersAdmin.saveCreateForm().click();
+
+    // we need to wait for that last API call to be finished to make sure we are ready for login
+    // with that new user
+    cy.wait('@globalRoleBindingCreated').then((res) => {
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/users-and-auth/users-and-auth.spec.ts:45]: would an `await` work here, or does `wait` not return something promise like?

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts (hunk 2 above)
Comment (aalves08) [reply cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts:37]: Updated. Thanks for the input

File: cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts (hunk 1 above)
Comment (aalves08) [reply cypress/e2e/po/pages/users-and-auth/users-and-auth.po.ts]: pattern matched

**Review by richard-cox — APPROVED** [review]

**Pull request merged**